clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -354,7 +355,8 @@
         "Requested position is not greater than or equal to zero, and less than the size of the snapshot.",
         PositionOutOfRangeException::new),
     UNKNOWN_TOPIC_ID(100, "This server does not host this topic ID.", UnknownTopicIdException::new),
-    DUPLICATE_BROKER_REGISTRATION(101, "This broker ID is already in use.", DuplicateBrokerRegistrationException::new);
+    DUPLICATE_BROKER_REGISTRATION(101, "This broker ID is already in use.", DuplicateBrokerRegistrationException::new),
+    INVALID_CLUSTER_ID(102, "The supplied cluster id is not valid.", InvalidClusterIdException::new);
```

Comment:
I am considering if we should have a more explicit name here. Specifically we are checking that the clusterId matches. Maybe `INCONSISTENT_CLUSTER_ID` would be clearer?

Reply:
I see your point. `INVALID` is mainly used by all the other errors so it also makes sense to follow the "naming convention". I don't feel strong either ways.

Reply:
I couldn't find any existing `INVALID*` error code that seems to fit this case. Usually "invalid" is reserved for cases where the field is structurally invalid. For example, `INVALID_GROUP_ID` is used when the groupid is empty. The closest similar case is `INVALID_PRODUCER_ID_MAPPING`. 

We are going to add an INCONSISTENT_TOPIC_ID in https://github.com/apache/kafka/pull/10143. Perhaps that is  enough cover here? The usage is similar: the request indicates an id which does not match the local state.